src/core_plugins/tile_map/public/maps_visualization.js

```diff
@@ -272,10 +272,40 @@ export function MapsVisualizationProvider(serviceSettings, Notifier, getAppState
 
       const indexPatternName = agg.vis.indexPattern.id;
       const field = agg.fieldName();
-      const filter = { meta: { negate: false, index: indexPatternName } };
-      filter[filterName] = { ignore_unmapped: true };
-      filter[filterName][field] = filterData;
-      getAppState().$newFilters = [filter];
+      const query = this.vis.API.queryManager.getQuery();
+      const language = query.language;
+
+      if (language === 'lucene') {
+        const filter = { meta: { negate: false, index: indexPatternName } };
+        filter[filterName] = { ignore_unmapped: true };
+        filter[filterName][field] = filterData;
+
+        this.vis.API.queryFilter.addFilters([filter]);
+      }
+      else if (language === 'kuery') {
+        const { fromKueryExpression, toKueryExpression, nodeTypes } = this.vis.API.kuery;
+        let newQuery;
+
+        if (filterName === 'geo_bounding_box') {
+          newQuery = nodeTypes.function.buildNode('geoBoundingBox', field, _.mapKeys(filterData, (value, key) => _.camelCase(key)));
+        }
+        else if (filterName === 'geo_polygon') {
+          newQuery = nodeTypes.function.buildNode('geoPolygon', field, filterData.points);
+        }
+        else {
+          throw new Error(`Kuery does not support ${filterName} queries`);
+        }
+
+        const allQueries = _.isEmpty(query.query)
```

Comment:
this whole part below should probably be as simple as `this.vis.API.queryManager.addQuery(newQuery);`

Reply:
I do think there's room to simplify the method for combining two queries. I don't think it necessarily belongs in queryManager though. Different users of the API will want to combine queries in different ways. Some will want ANDs. Some will want ORs. Others will want even more business logic. I think an `add` method might make sense in the kuery node API but I'm not sure yet. Still figuring it out.